testing/README.md

```diff
@@ -277,3 +277,214 @@ with the `BenchmarkRunOptions.silent` flag.
+
+  // asserts that multiplySpy was called at least once and details about the first call.
+  assertSpyCall(multiplySpy, 0, {
+    args: [5, 5],
+    returned: 25,
+  });
+
+  // asserts that multiplySpy was only called once.
+  assertSpyCalls(multiplySpy, 1);
+});
+```
+
+If you prefer not adding additional parameters for testing purposes only, you
+can use spy to wrap a method on an object instead. In the following example, the
+exported `_internals` object has the `multiply` function we want to call as a
+method and the `square` function calls `_internals.multiply` instead of
+`multiply`.
+
+```ts
+// https://deno.land/std@$STD_VERSION/testing/mock_examples/internals_injection.ts
+export function multiply(a: number, b: number): number {
+  return a * b;
+}
+
+export function square(value: number): number {
+  return _internals.multiply(value, value);
+}
+
+export const _internals = { multiply };
+```
+
+This way, we can call `square(value)` in both the application code and testing
+code. Then spy on the `multiply` method on the `_internals` object in the
+testing code to be able to spy on how the `square` function calls the `multiply`
+function.
+
+```ts
+// https://deno.land/std@$STD_VERSION/testing/mock_examples/internals_injection_test.ts
+import {
+  assertSpyCall,
+  assertSpyCalls,
+  spy,
+} from "https://deno.land/std@$STD_VERSION/testing/mock.ts";
+import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+import {
+  _internals,
+  square,
+} from "https://deno.land/std@$STD_VERSION/testing/mock_examples/internals_injection.ts";
+
+Deno.test("square calls multiply and returns results", () => {
+  const multiplySpy = spy(_internals, "multiply");
+
+  try {
+    assertEquals(square(5), 25);
+  } finally {
+    // unwraps the multiply method on the _internals object
+    multiplySpy.restore();
+  }
+
+  // asserts that multiplySpy was called at least once and details about the first call.
+  assertSpyCall(multiplySpy, 0, {
+    args: [5, 5],
+    returned: 25,
+  });
+
+  // asserts that multiplySpy was only called once.
+  assertSpyCalls(multiplySpy, 1);
+});
+```
+
+One difference you may have noticed between these two examples is that in the
+second we call the `restore` method on `multiplySpy` function. That is needed to
+remove the spy wrapper from the `_internals` object's `multiply` method. The
+`restore` method is called in a finally block to ensure that it is restored
+whether or not the assertion in the try block is successful. The `restore`
+method didn't need to be called in the first example because the `multiply`
+function was not modified in any way like the `_internals` object was in the
+second example.
+
+### Stubbing
+
+Say we have two functions, `randomMultiple` and `randomInt`, if we want to
+assert that `randomInt` is called during execution of `randomMultiple` we need a
+way to spy on the `randomInt` function. That could be done with either either of
```

Comment:
nit:
```suggestion
way to spy on the `randomInt` function. That could be done with either of
```

Reply:
This was a mistake I missed when typing this up. I can fix it tonight after work.

Reply:
This sentence wasn't fixed before this PR was merged. I'll fix this tonight in one of my PRs that haven't been merged yet that depend on this.